### PR TITLE
CalcOpticalFlowPyrLK added and GoodFeaturesToTrack fixed

### LINKED ISSUE #276: Any plans to add calcOpticalFlowPyrLK
Hi,

Is there any plan to add calcOpticalFlowPyrLK?

If someone could point me in the right direction I could maybe have a crack at it (I'm working with opencv 3)

Thanks

Carey

### PR DESCRIPTION
**Pull request by madshall**
`GoodFeaturesToTrack` was referring to hardcoded parameters, which are now being taken as method arguments.

`CalcOpticalFlowPyrLK` was added along with two constatnts: `cv.TERM_CRITERIA_EPS` and  `cv.TERM_CRITERIA_COUNT`

Also, `./examples/optical-flow.js` was added. 

This also closes #276 and responds to #486

### CODE REVIEW

File: src/OpenCV.cc [hunk 1]
```diff
@@ -9,6 +9,8 @@ void OpenCV::Init(Local<Object> target) {
   char out [21];
   int n = sprintf(out, "%i.%i", CV_MAJOR_VERSION, CV_MINOR_VERSION);
   target->Set(Nan::New<String>("version").ToLocalChecked(), Nan::New<String>(out, n).ToLocalChecked());
+  target->Set(Nan::New("TERM_CRITERIA_EPS").ToLocalChecked(), Nan::New<Integer>((int)cv::TermCriteria::EPS));
```
Comment (peterbraden) [inline src/OpenCV.cc]: Should these be here? Maybe in constants.cc?

Comment (madshall) [inline src/OpenCV.cc]: Good point.

**Comment by peterbraden** [comment]
Thanks!

**Comment by rtzm** [comment]
I was trying to run this example using a local Node install, and hit an error similar to this one: https://github.com/peterbraden/node-opencv/issues/498.

When I inspect the methods attached to Matrix.prototype in REPL, I don't see .calcOpticalFlowPyrLK() listed. See the list of available methods here: 

`
cv.Matrix.prototype.__defineGetter__
cv.Matrix.prototype.__defineSetter__
cv.Matrix.prototype.__lookupGetter__
cv.Matrix.prototype.__lookupSetter__
cv.Matrix.prototype.__proto__
cv.Matrix.prototype.constructor
cv.Matrix.prototype.hasOwnProperty
cv.Matrix.prototype.isPrototypeOf
cv.Matrix.prototype.propertyIsEnumerable
cv.Matrix.prototype.toLocaleString
cv.Matrix.prototype.toString
cv.Matrix.prototype.valueOf

cv.Matrix.prototype.absDiff
cv.Matrix.prototype.adaptiveThreshold
cv.Matrix.prototype.addWeighted
cv.Matrix.prototype.adjustROI
cv.Matrix.prototype.bilateralFilter
cv.Matrix.prototype.bitwiseAnd
cv.Matrix.prototype.bitwiseNot
cv.Matrix.prototype.bitwiseXor
cv.Matrix.prototype.brightness
cv.Matrix.prototype.canny
cv.Matrix.prototype.channels
cv.Matrix.prototype.clone
cv.Matrix.prototype.col
cv.Matrix.prototype.convertGrayscale
cv.Matrix.prototype.convertHSVscale
cv.Matrix.prototype.convertTo
cv.Matrix.prototype.copy
cv.Matrix.prototype.copyTo
cv.Matrix.prototype.copyWithMask
cv.Matrix.prototype.countNonZero
cv.Matrix.prototype.crop
cv.Matrix.prototype.cvtColor
cv.Matrix.prototype.detectObject
cv.Matrix.prototype.dilate
cv.Matrix.prototype.drawAllContours
cv.Matrix.prototype.drawContour
cv.Matrix.prototype.ellipse
cv.Matrix.prototype.empty
cv.Matrix.prototype.equalizeHist
cv.Matrix.prototype.erode
cv.Matrix.prototype.fillPoly
cv.Matrix.prototype.findContours
cv.Matrix.prototype.flip
cv.Matrix.prototype.floodFill
cv.Matrix.prototype.gaussianBlur
cv.Matrix.prototype.get
cv.Matrix.prototype.getData
cv.Matrix.prototype.getPerspectiveTransform
cv.Matrix.prototype.goodFeaturesToTrack
cv.Matrix.prototype.height
cv.Matrix.prototype.houghCircles
cv.Matrix.prototype.houghLinesP
cv.Matrix.prototype.inRange
cv.Matrix.prototype.inspect
cv.Matrix.prototype.line
cv.Matrix.prototype.locateROI
cv.Matrix.prototype.matchTemplate
cv.Matrix.prototype.matchTemplateByMatrix
cv.Matrix.prototype.mean
cv.Matrix.prototype.meanStdDev
cv.Matrix.prototype.meanWithMask
cv.Matrix.prototype.medianBlur
cv.Matrix.prototype.merge
cv.Matrix.prototype.minMaxLoc
cv.Matrix.prototype.moments
cv.Matrix.prototype.norm
cv.Matrix.prototype.normalize
cv.Matrix.prototype.pixel
cv.Matrix.prototype.pixelCol
cv.Matrix.prototype.pixelRow
cv.Matrix.prototype.ptr
cv.Matrix.prototype.pushBack
cv.Matrix.prototype.put
cv.Matrix.prototype.putText
cv.Matrix.prototype.pyrDown
cv.Matrix.prototype.pyrUp
cv.Matrix.prototype.rectangle
cv.Matrix.prototype.release
cv.Matrix.prototype.reshape
cv.Matrix.prototype.resize
cv.Matrix.prototype.roi
cv.Matrix.prototype.rotate
cv.Matrix.prototype.row
cv.Matrix.prototype.save
cv.Matrix.prototype.saveAsync
cv.Matrix.prototype.set
cv.Matrix.prototype.setWithMask
cv.Matrix.prototype.shift
cv.Matrix.prototype.size
cv.Matrix.prototype.sobel
cv.Matrix.prototype.split
cv.Matrix.prototype.subtract
cv.Matrix.prototype.templateMatches
cv.Matrix.prototype.threshold
cv.Matrix.prototype.toBuffer
cv.Matrix.prototype.toBufferAsync
cv.Matrix.prototype.warpAffine
cv.Matrix.prototype.warpPerspective
cv.Matrix.prototype.width
`

**Comment by rtzm** [comment]
It looks like the issue was that this never made its way into the version published on NPM. I'm using the github version for now.

**Pull request merged**